components/TodoList.js

```diff
@@ -105,55 +88,69 @@ function TodoList() {
         id: Math.random().toString(32).substring(2),
         text: newTodo,
         priority: priority,
+        completed: false,
       };
       setTodos([...todos, todoItem]);
       setNewTodo('');
     }
   };
 
   const handleRemoveTodo = (id) => {
-    const confirmDelete = window.confirm('Are you sure you want to delete this task?');
+    const confirmDelete = window.confirm('Bist du sicher dass du es löschen möchtest ');
     if (confirmDelete) {
       const newTodos = todos.filter((todo) => todo.id !== id);
       setTodos(newTodos);
     }
   };
 
   const handleAddNote = (id, note) => {
-    const noteWords = note.split(' ');
-
-    if (noteWords.length <= 1 && note.length <= 12) {
-      const truncatedNote = noteWords.slice(0, 6).join(' ');
-      setNotes((prevNotes) => ({ ...prevNotes, [id]: truncatedNote }));
+    if (note && note.trim() !== '') { 
+      const noteWords = note.split(' ');
+  
+      if (noteWords.length <= 1 && note.length <= 12) {
+        const truncatedNote = noteWords.slice(0, 6).join(' ');
+        setNotes((prevNotes) => ({ ...prevNotes, [id]: truncatedNote }));
+      } else {
+        alert('Maximal 12 Zeichen sind für Notizen erlaubt.');
+      }
     } else {
-      alert('Maximal 12 Zeichen sind für Notizen erlaubt.');
+      alert('Bitte geben Sie eine gültige Notiz ein.');
```

Comment:
there's a little issue with the "abbrechen" button, when i'm clicking at the "notes" in any task and press the "abbrechen" button, the alert appears. maybe you can find a way to run only in this alert, when the "ok" button with no note will be pressed.

Reply:
i know it's a bit harder but it will help you in the future!